### PR TITLE
Improve inference export script

### LINKED ISSUE #3038: Improve inference export script
`inference/server/export.py`

- Add ability to filter by conditions
- Include a filter for only chats which have at least one assistant message with score != 0
- Include a filter for specific date ranges
- Consider performance improvements such as use of lazy loading from the database

### PR DESCRIPTION
**Pull request by simonusher**
closes #3038

### CODE REVIEW

File: inference/server/export.py [hunk 1]
```diff
@@ -219,19 +226,68 @@ def parse_args() -> argparse.Namespace:
...
+        help="Only export this chat.",
+    )
+    parser.add_argument(
+        "--nonzero-assistant-score",
+        action=argparse.BooleanOptionalAction,
+        help="Only export chats with at least one assistant message with score != 0.",
+    ),
+    parser.add_argument(
+        "--reported",
+        action=argparse.BooleanOptionalAction,
```
Comment (olliestanley) [inline inference/server/export.py:256]: Could we use `action="store_true"` for these filters? Would just keep it consistent with other boolean args for this and other scripts

**Review by olliestanley — COMMENTED** [review]
Thanks for this PR, changes look good, I have left some minor comments and will run it locally to test soon

File: inference/server/export.py [hunk 2]
```diff
@@ -163,13 +164,16 @@ def write_messages_to_file(
...
 
 
-async def fetch_eligible_chats(session_generator) -> list[DbChat]:
-    """Fetch chats which are not opted out of data collection."""
+async def fetch_eligible_chats(
+    session_generator, filters: list[sqlalchemy.sql.elements.BinaryExpression]
+) -> list[DbChat]:
+    """Fetch chats which are not opted out of data collection and match the given filters."""
     session: AsyncSession
+    filters.append(DbChat.allow_data_use)
```
[the remark this answers is not in the document]
Comment (simonusher) [reply inference/server/export.py:170]: Sure, but I don't see a 'deleted' field. Do you mean 'hidden'?

[the remark this answers is not in the document]
Comment (olliestanley) [reply inference/server/export.py:170]: Sorry, ignore this, I forgot that in inference we went with hard deletion (so the row is just dropped from the table) whereas in data collection backend we soft delete with a flag instead. My bad

**Comment by olliestanley** [comment]
> I'm glad to hear that. I've addressed your comments except for one. Please let me know if the performance is sufficient. I believe the relationship filters, as they are now, will use correlated subqueries so they might be suboptimal. I'm not entirely sure how to test them properly at scale. Is there maybe a script that fills the development inference-db with dummy data?

There isn't a way of filling the inference DB with dummy data currently, this would be a useful addition though so I'll raise a new issue for it. There's some dummy data [here](https://github.com/LAION-AI/Open-Assistant/tree/main/backend/test_data) which was used for filling the data collection backend DB, could maybe be reused

**Review by olliestanley — APPROVED** [review]

**Comment by olliestanley** [comment]
Thank you for this PR!

**Pull request merged**